### PR TITLE
Update Residue.py

### PR DESCRIPTION
**Pull request by rodogi**
get_atoms method seems more in touch with the methods used for chain, model and structure.

### CODE REVIEW

**Comment by peterjc** [comment]
You can't just remove an old API like that, it would break any script using the old function name. We'd have to have both in parallel for a while and deprecate the old name.

However, there was talk of deprecating all the PDB module's ``get_*`` methods in preference to using Python properties (e.g. ``.atoms`` instead of ``.get_atoms()`` as so on).

**Comment by JoaoRodrigues** [comment]
While I fully agree with Peter's first sentence, this seems more of a typo
than anything else actually. I *never* noticed this and I've been using
this module for years.. I'd keep an alias to the function and add the
deprecation warning actually.​

**Comment by peterjc** [comment]
So we're agreed :) - rename it but keep a backward compatible alias with a deprecation warning when used. See also http://biopython.org/wiki/Deprecation_policy

**Comment by rodogi** [comment]
I added the new method `get_atoms` while keeping the old one `get_atom` with a deprecation warning.

**Comment by JoaoRodrigues** [comment]
Looks good to me. Thanks @rodogi for spotting & fixing this.

**Comment by peterjc** [comment]
The change looks great 👍 

Are you happy to dual licence your contributions under both the _Biopython License Agreement_ AND the _BSD 3-Clause License_? See #898.

Also, are you happy to be thanked by name in the ``NEWS.rst`` and ``CONTRIB.rst`` files?

Does this currently get triggered by any of the unit tests?

**Comment by rodogi** [comment]
Thank you guys for your terrific work in biopython. 

> Does this currently get triggered by any of the unit tests?

I don't really know, I'm still a noob to github.

**Comment by peterjc** [comment]
It looks like the old name does get used in ``test_mmtf.py`` (only):

```
$ grep "\.get_atom(" *.py
test_mmtf.py:            self.mmcif_atoms = [x for x in mmcif_r.get_atom()]
test_mmtf.py:            self.mmtf_atoms = [x for x in mmtf_r.get_atom()]
```

This is confirmed in the test suite output on TravisCI where the new deprecation warning appears:

```
...
test_mmtf ... /home/travis/build/biopython/biopython/build/lib.linux-x86_64-2.7/Bio/PDB/Residue.py:120: BiopythonDeprecationWarning: `get_atom` has been depcreated and we intend to remove it in a future release of Biopython. Please use `get_atoms` instead.
  BiopythonDeprecationWarning)
ok
...
```

So, could you also update ``test_mmtf.py`` to use the new ``.get_atoms()`` version please. With that I'd be happy to merge this pull request.

Separately this shows the old method was not being deliberately tested in ``test_PDB.py``, and it (or rather the new version) ought to be. If you'd like to tackle that as well @rodogi that would be most welcome.

File: Bio/PDB/Residue.py [hunk 1]
```diff
@@ -108,7 +110,14 @@ def get_unpacked_list(self):
     def get_segid(self):
         return self.segid
 
+    def get_atoms(self):
+        for a in self:
+            yield a
+
     def get_atom(self):
+        warnings.warn("`get_atom` has been depcreated and we intend to remove it"
```
Comment (peterjc) [inline Bio/PDB/Residue.py]: Typo: ``depcreated`` --> ``deprecated``

**Comment by peterjc** [comment]
@rodogi How would you like to be named in the ``NEWS.rst`` and ``CONTRIB.rst`` files? I'm asking as you don't make this obvious on your GitHub profile, perhaps deliberately. Should you wish to remain semi-anonymous, just ``rodogi`` is fine (there is recent precedent).

You can add yourself to these files as part of the pull request if you like - note the names are sorted alphabetically and for ``CONTRIB.rst`` this is checked automatically as described in the header.

**Comment by peterjc** [comment]
Thank you 👍

**Comment by peterjc** [comment]
@rodogi Are you happy to dual licence your contributions under both the _Biopython License Agreement_ AND the _BSD 3-Clause License_? See #898.

**Comment by rodogi** [comment]
@peterjc Yes I agree to dual licence my contributions under both the Biopython License Agreement AND the BSD 3-Clause License.

**Comment by peterjc** [comment]
@rodogi Thank you 👍

**Pull request merged**